packages/ui/src/components/Carousel/Carousel.tsx

```diff
@@ -88,11 +88,15 @@ export const Carousel: FC<CarouselProps> = ({
 
   const items = useMemo(
     () =>
-      Children.map(children as ReactElement[], (child: ReactElement) =>
-        cloneElement(child, {
-          className: twMerge(theme.item.base, child.props.className),
-        }),
-      ),
+      Children.map(children as ReactElement[], (child: ReactElement<CarouselProps>) => {
+        if (isValidElement(child)) {
+          return cloneElement(child as ReactElement, {
+            className: twMerge(theme.item.base, child.props.className || ""),
+          });
+        }
+        // If child is not a valid React element, return it as-is or handle it accordingly
+        return child;
```

Comment:
Don't we also want to include the `twMerge` for the `child.className` here? :thinking:

Reply:
but here, on this specific line there won't be any `child.className` because of the dynamic child, `child.props` & `child.props.className` is throwing Error @rluders

Reply:
@rluders & @SutuSebastian --- your inputs please